tools/Google.Cloud.Tools.ConstructorGenerator/Program.cs

```diff
@@ -0,0 +1,128 @@
+            foreach (var api in apis)
+            {
+                GenerateConstructors(api);
+            }
+        }
+
+        static void GenerateConstructors(ApiMetadata api)
+        {
+            Console.WriteLine($"Processing {api.Id}");
+            var asm = PublishUnitTestsAndLoadAssembly(api);
+            var clientTypes = asm.GetTypes().Where(t => t.IsClass && t.BaseType.IsConstructedGenericType && t.BaseType.GetGenericTypeDefinition() == typeof(ClientBase<>)).ToList();
+
+            if (clientTypes.Count > 0)
+            {
+                var lines = new List<string>
+                {
+                    "// Copyright 2020 Google LLC",
+                    "//",
+                    "// Licensed under the Apache License, Version 2.0 (the \"License\");",
+                    "// you may not use this file except in compliance with the License.",
+                    "// You may obtain a copy of the License at",
+                    "//",
+                    "//     https://www.apache.org/licenses/LICENSE-2.0",
+                    "//",
+                    "// Unless required by applicable law or agreed to in writing, software",
+                    "// distributed under the License is distributed on an \"AS IS\" BASIS,",
+                    "// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.",
+                    "// See the License for the specific language governing permissions and",
+                    "// limitations under the License.",
+                    "",
+                    "// Partial class constructors to preserve compatibility when migrating",
+                    "// from gRPC 1.x to 2.x tooling.",
+                    "",
+                    "using Grpc.Core;",
+                    "using System;",
+                    "",
+                    $"namespace {api.Id}",
+                    "{"
+                };
+
+                foreach (var type in clientTypes)
+                {
+                    var name = type.Name[0..^6]; // Strip off client
+                    lines.AddRange(new[]
+                    {
+                        // All interpolated to make indentation easier to verify
+                        $"    public partial class {name}",
+                        $"    {{",
+                        $"        public partial class {name}Client",
+                        $"        {{",
+                        $"            /// <summary>Creates a new client using a channel.</summary>",
+                        $"            /// <param name=\"channel\">The channel to use to make remote calls.</param>",
+                        $"            [Obsolete(\"This constructor overload is present for compatibility only, and will be removed in the next major version\")]",
+                        $"            public {name}Client(Channel channel) : base(channel)",
```

Comment:
Not sure if it should be so that it calls the new constructor. If the new one is now doing stuff that this one did before, then I think this suggestion stands.

```suggestion
                        $"            public {name}Client(Channel channel) : this(channel)",
```

Reply:
The previous constructor had `base(channel)` so it's really just doing exactly the same as before the change. (And the other constructor also just uses `base(channel)`, so it makes no odds.)